
syntax = "proto3";

package user.v1;

import "google/api/annotations.proto";
import "protoc-gen-openapiv2/options/annotations.proto";

option go_package = "github.com/klaital/cardsorter/backend/gen/protos;cardv1";

option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
  info: {
    title: "User Service API";
    version: "1.0";
    description: "API for managing users and sessions";
    contact: {
      name: "Login Service Team";
      url: "https://github.com/klaital/cardsorter";
    };
  }
};

service UserService {
  // CreateUser registers a new user
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {
    option (google.api.http) = {
      post: "/v1/users"
      body: "*"
    };
  }

  // Login authenticates a user and returns a token
  rpc Login(LoginRequest) returns (LoginResponse) {
    option (google.api.http) = {
      post: "/v1/login"
      body: "*"
    };
  }
}

message CreateUserRequest {
  string email = 1;
  string password = 2;
}

message CreateUserResponse {
  int64 id = 1;
  string token = 2;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  int64 user_id = 1;
  string token = 2;
}